pkg/deploy/generator/resources_gateway.go

```diff
@@ -421,5 +422,8 @@ func (g *generator) gatewayRBAC() []*arm.Resource {
 }
 
 func (g *generator) gatewayStorageAccount() *arm.Resource {
-	return g.storageAccount("[substring(parameters('gatewayStorageAccountDomain'), 0, indexOf(parameters('gatewayStorageAccountDomain'), '.'))]", nil, nil)
+	accountProperties := mgmtstorage.AccountProperties{
```

Comment:
Duplicate

Reply:
Updated the PR making this change in a way that is more clean and reusable, as per our discussion on Teams. Thanks!